Firestore/core/src/api/firestore.cc

```diff
@@ -242,6 +251,68 @@ DatabaseInfo Firestore::MakeDatabaseInfo() const {
                       settings_.ssl_enabled());
 }
 
+void Firestore::SetIndexConfiguration(std::string config,
+                                      util::StatusCallback callback) {
+  EnsureClientConfigured();
+
+  bundle::JsonReader reader;
+  if (!settings_.persistence_enabled()) {
+    callback(Status(Error::kErrorFailedPrecondition,
+                    "Cannot enable indexes when persistence is disabled."));
+    return;
+  }
+
+  auto json_object =
+      nlohmann::json::parse(config.begin(), config.end(),
+                            /*callback=*/nullptr, /*allow_exceptions=*/false);
+  if (json_object.is_discarded()) {
+    callback(Status(Error::kErrorDataLoss, "Invalid Json format."));
```

Comment:
I see you copied `kErrorDataLoss` from bundles, but I still think invalid argument makes more sense in this context.